config-provisioning/src/main/java/com/yahoo/config/provision/AllocatedHosts.java

```diff
@@ -54,14 +64,35 @@ private void toSlime(Cursor cursor) {
     }
 
     private void toSlime(HostSpec host, Cursor cursor) {
-        cursor.setString(hostSpecHostName, host.hostname());
+        cursor.setString(hostSpecHostNameKey, host.hostname());
+        aliasesToSlime(host, cursor);
         host.membership().ifPresent(membership -> {
-            cursor.setString(hostSpecMembership, membership.stringValue());
-            cursor.setString(hostSpecVespaVersion, membership.cluster().vespaVersion().toFullString());
+            cursor.setString(hostSpecMembershipKey, membership.stringValue());
+            cursor.setString(hostSpecVespaVersionKey, membership.cluster().vespaVersion().toFullString());
         });
-        host.flavor().ifPresent(flavor -> cursor.setString(hostSpecFlavor, flavor.name()));
-        host.version().ifPresent(version -> cursor.setString(hostSpecCurrentVespaVersion, version.toFullString()));
-        host.networkPorts().ifPresent(ports -> NetworkPortsSerializer.toSlime(ports, cursor.setArray(hostSpecNetworkPorts)));
+        host.flavor().ifPresent(flavor -> toSlime(flavor, cursor));
+        host.version().ifPresent(version -> cursor.setString(hostSpecCurrentVespaVersionKey, version.toFullString()));
+        host.networkPorts().ifPresent(ports -> NetworkPortsSerializer.toSlime(ports, cursor.setArray(hostSpecNetworkPortsKey)));
+    }
+
+    private void aliasesToSlime(HostSpec spec, Cursor cursor) {
+        if (spec.aliases().isEmpty()) return;
+        Cursor aliases = cursor.setArray(aliasesKey);
+        for (String alias : spec.aliases())
+            aliases.addString(alias);
+    }
+
+    private void toSlime(Flavor flavor, Cursor object) {
+        if (flavor.isConfigured()) {
+            object.setString(flavorKey, flavor.name());
```

Comment:
Isn't this the same problem with had last time? Why not just always write flavor, then on the version it can be moved inside the `if`?

Reply:
Because the "names" of non-configured flavors cannot be parsed back to flavors.
Here we should just remove the flavor part once we have no bare metal left (contrary to the node repo which stores both hosts and nodes).

Reply:
Looks like deserialization would work either way. If we always write the name, the old deserializer code would just return `Optional.empty()` since `NodeFlavors` would not know about non-configured flavor. If we don't write it, then I guess `object.field(hostSpecFlavor).valid()` is `false` so that works too. :+1:

Reply:
Yep  :-)